plugins/org.eclipse.glsp.server/src/org/eclipse/glsp/server/operations/gmodel/CreateNodeOperationHandler.java

```diff
@@ -40,13 +41,13 @@ public void executeOperation(final CreateNodeOperation operation, final GModelSt
          container = Optional.of(modelState.getRoot());
       }
 
-      GModelElement element = createNode(getLocation(operation), modelState);
+      GModelElement element = createNode(getLocation(operation), operation.getArgs(), modelState);
```

Comment:
As far as I can see the map may be null here. I am not too much of a fan of null collection parameters, especially when point is wrapped in a nice Optional. But it could be that this is consistent with other operation handlers as well, just wanted to mention that in case it is different anywhere else.

Reply:
Good point. I updated the `CreateNodeOperation` so that the args map is initalized with an empty map. This way we can avoid the overhead of wrapping the Map into an `Optional` and the map should always be defined. Unless somebody explicitly uses the setter to set the map to null, which should hopefully never be the case :wink:)

Reply:
I like having an empty map before having an optional map much more! However, it seems that CreateEdgeOperation and CreateNodeOperation both initialize the `args` with null due to re-using the full constructor. So either we do a `putAll` in the `CreateNodeOperation` constructor for non-null maps OR the subclasses also need to use an empty map as their default constructor argument.

Reply:
Good catch. I checked other occurrences where we use an arbitrary args Map (e.g. EditorContext, RequestModelAction etc.) and we always more or less operate on the assumption that this map is not null. So I updated the constructors of all classes that use an args Map to ensure that they are initialized with an empty map.